### ISSUE TITLE
gRPC server instrumentation can not link grpc request transaction

### ISSUE THREAD

**Issue by mashiro** [newrelic/newrelic-java-agent#942]
## Description
When gRPC instrumentation is used, the token is not active in the server method and cannot be linked by asynchronous operation.
Also, if `@Trace(dispatcher = true)` is given in the server method, subsequent segments of asynchronous operations will not be recorded as transactions.
This will work correctly if grpc-instrumentation is disabled.

## Expected Behavior
token is active in gRPC server method.

## Your Environment
* gRPC 1.47.0
* NewRelic Agent 7.8.0

(Migrate to Jira)

**Comment by kford-newrelic** [comment]
@mashiro thanks for letting us know about this issue

Any chance you have a sample app/code that reproduces this issue?

**Comment by mashiro** [comment]
Hi, @kford-newrelic 
I use grpc-java example code. https://grpc.io/docs/languages/java/quickstart/#get-the-example-code

1. Implement debug code to HelloWorldServer.java
```
Token token = NewRelic.getAgent().getTransaction().getToken();
System.out.println("token.isActive() = " + token.isActive());
```
2. Download newrelic agent to example directory
3. Run hello-world-server with newrelic-agent
```
HELLO_WORLD_SERVER_OPTS="-javaagent:../../../../newrelic/newrelic.jar" ./hello-world-server
```
4. Run hello-world-client
```
./hello-world-client
```

**Comment by kford-newrelic** [comment]
Targeted for the Jan-Mar quarter

**Comment by obenkenobi** [comment]
The bug is already fixed for the upcoming 8.0 release so I am closing off the issue.

**Comment by obenkenobi** [comment]
For reference, this issue is a duplicate of [another issue](https://github.com/newrelic/newrelic-java-agent/issues/1023) which already closed out.
